fmf/base.py

```diff
@@ -588,3 +590,80 @@ def copy(self):
         duplicate = copy.deepcopy(self)
         self.parent = duplicate.parent = original_parent
         return duplicate
+
+    def __get_raw_data_node(self):
+        """
+        Get tuple of node where raw_data are stored and list if keys
+        for current node inside raw data
+        """
+        raw_data_node = self
+        id_list = list()
+        while True:
+            raw_data = raw_data_node.raw_data
+            if raw_data:
+                return raw_data_node, id_list
+            elif not raw_data_node.parent:
+                raise utils.RootError("Cannot find root node with raw_data, very strange")
+            else:
+                id_list.insert(0, "/" + raw_data_node.name.rsplit("/")[-1])
+                raw_data_node = raw_data_node.parent
+
+    def __get_node_raw_data(self):
+        """
+        Get dictionary of raw_data for current node
+        """
+        raw_data_node, id_list = self.__get_raw_data_node()
+        raw_data = raw_data_node.raw_data
+        for key in id_list:
+            raw_data = raw_data[key]
+        return raw_data
+
+    def modify(self, data, method="update", reload=False):
+        """
+        Modify node metadata (original data stored in file) via method.
+        method is any method what contains python dictionary (default: update)
+        and data is attribute of the method
+
+        It stores data back to files.
+
+        set reload  to True to dynamically recreate the subtree,
+        theoretically it may have some side effects, use it carefully.
+        Or reload whole tree to ensure data are completely fine.
+        """
+
+        getattr(self.__get_node_raw_data(), method)(data)
+        self.__store()
+        if reload:
+            self.reload()
+        return self
+
+    def __data_cleanup(self):
+        for item in self.climb(whole=True):
+            item.data = dict()
+            item.sources = []
+
+    def reload(self):
+        """
+        Try to reload data after it is modified
+        It reloads data from node where are defined.
+
+        WARNING:
+        This method may lead that some variables to nodes in your code could lost references.
+        In case of bigger modification, it would be better to reload whole tree
+        in your code.
```

Comment:
Well, I'm not completely sure if we should provide a functionality which works only maybe or sometimes... If this is too complex, shall we leave it for later and if really needed? Anyway, why is `reload` needed?